### ISSUE TITLE
[Http-Client] 

### ISSUE THREAD

**Issue by Iyadhfaleh** [symfony/symfony#46499]
### Symfony version(s) affected

4.4.39

### Description

We faced a bug after upgrading Http-client from 4.4.35 to  >= 4.4.39
 

### How to reproduce

We are using this configuration of our service :

` our.client:
                base_uri: 'https://lorem-ipsum.com'
                proxy: 'http://proxy.net:8080'
                headers:
                    Accept: 'application/json'`
                    
 And we make calls like bellow :
 `$response = $this->client->request(Request::METHOD_POST, 'token', [
                'body' => [
                    'username' => $userName,
                    'password' => $password,
                    'grant_type' =>password,
                    'client_id' => $clientId,
                    'client_secret' => $client_secret,
                ],
            ]);
            $result = $response->toArray();`
            
We received this exception `Failure when receiving data from the peer for ...`

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by Iyadhfaleh** [comment]
@xabbuh 

In services.yaml , we configured it like this :

```   http_client:
        scoped_clients:
              our_client.client:
                base_uri: 'https://lorem-ipsum.com' 
                proxy: 'http://proxy.net:8080'
                headers: Accept: 'application/json'
        ```
        
     And we make calls like this : 
     
     ```
        $response = $this->ourClientClient->request(Request::METHOD_POST, 'token', [
                'body' => [
                    'username' => $userName,
                    'password' => $password,
                    'grant_type' => 'password',
                    'client_id' => $clientId,
                    'client_secret' => $clientSecret
                ],
            ]);

            $result = $response->toArray();
            ```

 PS : with native curl calls works without problems

**Comment by xabbuh** [comment]
I am afraid one needs a full example application that one can just clone and run to reproduce the error.

**Comment by nicolas-grekas** [comment]
Sending option `body` without a `Content-Type` means you're sending the data as `application/x-www-form-urlencoded`. Is this really what your API expects? Can you figure out the exact version or even commit that introduced the change for you?

**Comment by xabbuh** [comment]
I am going to close here for now due to the lack of feedback. Please let us know when you have more information and we can consider to reopen.

**Comment by nicolas-grekas** [comment]
Should be fixed by #48966. If you have the opportunity, please give it a try and report back.
